File: web_login_flutter/lib/models/form/section.dart
import 'package:equatable/equatable.dart';
import 'package:web_login_flutter/models/form/question.dart';

class Section extends Equatable {
  final int _id;
  final String _name;
  final String? _description;
  final int _totalQuestions;
  final Question _question;

  const Section({required id,
    required name,
    required description,
    required totalQuestions,
    required question}) :
        _id = id,
        _name = name,
        _totalQuestions = totalQuestions,
        _description = description,
        _question = question;


  @override
  List<Object?> get props => [_id, _name, _totalQuestions, _description, _question];

  @override
  String toString() {
    return 'Section{_id: $_id, _name: $_name, _description: $_description, _totalQuestions: $_totalQuestions, _question: $_question}';
  }

  int get id => _id;

  Question get question => _question;

  int get totalQuestions => _totalQuestions;

  String? get description => _description;

  String get name => _name;
}

File: web-login-core/src/main/java/com/example/weblogincore/domain/model/form/Section.java
package com.example.weblogincore.domain.model.form;

public class Section {
    private final Long id;
    private final String name;
    private final String description;

    private final Question question;

    public Section(Long id, String name, String description, Question question) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.question = question;
    }

    public Integer totalQuestions() {
        return question.totalQuestions();
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public Question getQuestion() {
        return question;
    }
}
